packages/core/src/new-api/types/module.ts

```diff
@@ -295,6 +296,48 @@ export type ModuleParameterType =
   | ModuleParameterType[]
   | { [field: string]: ModuleParameterType };
 
+/**
+ * The different runtime values supported by Ignition.
+ *
+ * @beta
+ */
+export enum RuntimeValueType {
+  ACCOUNT = "ACCOUNT",
+  MODULE_PARAMETER = "MODULE_PARAMETER",
+}
+
+/**
+ * A value that's only available during deployment.
+ *
+ * @beta
+ */
+export type RuntimeValue =
+  | AccountRuntimeValue
+  | ModuleParameterRuntimeValue<ModuleParameterType>;
+
+/**
+ * A local account.
+ *
+ * @beta
+ */
+export interface AccountRuntimeValue {
+  type: RuntimeValueType.ACCOUNT;
+  accountIndex: number;
+}
+
+/**
+ * A module parameter.
+ *
+ * @beta
+ */
+export interface ModuleParameterRuntimeValue<
+  ParamTypeT extends ModuleParameterType
```

Comment:
This can't receive any `ArgumentType`, for example, it can't receive a `Future`. That's why we need a different type.